src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionSymbolService.cs

```diff
@@ -30,5 +39,22 @@ public async Task<(ISymbol, TextSpan)> GetSymbolAndBoundSpanAsync(Document docum
 
             return (FindRelatedExplicitlyDeclaredSymbol(symbol, semanticModel.Compilation), semanticInfo.Span);
         }
+
+        private async Task<SyntaxToken> GetToken(SemanticModel semanticModel, Workspace workspace, int position, CancellationToken cancellationToken)
```

Comment:
I think you pulled this code out of `SymbolFinder.GetSemanticInfoAtPositionAsync`? That method is also used by "peek definition" (alt-f12) so it would be nice to get the same fix made to that feature as we are making to Go To Definition and Ctrl-Click GTD. Can you instead add a new method to ISyntaxFactsService "IsPropertyAccessorToken" (or something), implement that in C# and VB, and then do the check in SymbolFinder?

Reply:
(and obviously, remove this and ShouldTryNavigateToToken from GoToDefinitionSymbolService)